casper/src/test/scala/coop/rchain/casper/util/rholang/InterpreterUtilTest.scala

```diff
@@ -30,40 +29,53 @@ import monix.eval.Task
 import monix.execution.Scheduler.Implicits.global
 import org.scalatest._
 
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Random
+
 class InterpreterUtilTest
     extends FlatSpec
     with Matchers
     with BlockGenerator
     with BlockDagStorageFixture {
+  import BlockGenerator.step
 
   implicit val logEff                    = new LogStub[Task]
   implicit val metricsEff: Metrics[Task] = new metrics.Metrics.MetricsNOP[Task]
   implicit val span: Span[Task]          = new NoopSpan[Task]
   implicit val logSource: LogSource      = LogSource(this.getClass)
+  implicit private val timeEff: Time[Task] = new Time[Task] {
+    private val constantTime                                 = 4L
+    override val currentMillis: Task[Long]                   = Task.pure(constantTime)
+    override val nanoTime: Task[Long]                        = Task.pure(constantTime)
+    override def sleep(duration: FiniteDuration): Task[Unit] = Task.sleep(duration)
+  }
 
   val genesisContext = GenesisBuilder.buildGenesis()
   val genesis        = genesisContext.genesisBlock
 
-  def computeDeploysCheckpoint[F[_]: Sync: Log: BlockStore: Span](
+  def computeDeploysCheckpoint[F[_]: Sync: Log: BlockStore: Span: Time](
       parents: Seq[BlockMessage],
       deploys: Seq[Signed[DeployData]],
       dag: BlockDagRepresentation[F],
       runtimeManager: RuntimeManager[F]
   ): F[Either[Throwable, (StateHash, StateHash, Seq[ProcessedDeploy])]] =
-    InterpreterUtil
-      .computeDeploysCheckpoint[F](
-        parents,
-        deploys,
-        dag,
-        runtimeManager,
-        BlockData(
-          deploys.maxBy(_.data.timestamp).data.timestamp,
-          0,
-          genesisContext.validatorPks.head
-        ),
-        Map.empty[BlockHash, Validator]
+    Time[F].currentMillis >>= (
+        now =>
+          InterpreterUtil
+            .computeDeploysCheckpoint[F](
+              parents,
+              deploys,
+              dag,
+              runtimeManager,
+              BlockData(
+                now,
```

Comment:
This does the job for now but does not reflect the reality where timestamps would increase from block to block. On the other hand smart contracts can't rely on the correctness of timestamp data anyway so having a constant time in tests might be a good thing.